src/extension/intellisenseHelper.ts

```diff
@@ -7,12 +7,25 @@ import * as os from "os";
 import * as path from "path";
 import * as Q from "q";
 import * as vscode from "vscode";
+import * as semver from "semver";
 import {Telemetry} from "../common/telemetry";
 import {TelemetryHelper} from "../common/telemetryHelper";
 import {TsConfigHelper} from "./tsconfigHelper";
 import {SettingsHelper} from "./settingsHelper";
+import {Log, LogLevel} from "../common/log";
+
+interface InstallProps {
+    installed: boolean;
+    version: string;
+}
 
 export class IntellisenseHelper {
+
+    private static s_typeScriptVersion = "1.8.2";           // preferred version of TypeScript for legacy VSCode installs
+    private static s_vsCodeVersion = "0.10.10-insider";     // preferred version of VSCode (current is 0.10.9, 0.10.10-insider+ will include native TypeScript support)
+                                                            // note: semver considers "x.x.x-<string>" to be < "x.x.x"" - so we include insider here as the
```

Comment:
An easy way to fix tslint erros as you go is to install the `tslint` vscode extension. It helps me save a lot of time before checking in.